amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -115,9 +134,28 @@ private void cleanTestResources(StreamExistenceManager streamExistenceManager, L
     }
 
     private void startProducer() {
-        // Send dummy data to stream
-        this.producerExecutor = Executors.newSingleThreadScheduledExecutor();
-        this.producerFuture = producerExecutor.scheduleAtFixedRate(this::publishRecord, 60, 1, TimeUnit.SECONDS);
+        producerExecutor = Executors.newSingleThreadScheduledExecutor();
+        producerFuture = producerExecutor.scheduleAtFixedRate(this::publishRecord, 10, 1, TimeUnit.SECONDS);
+
+        // Reshard logic if required for the test
+        if (consumerConfig.getReshardConfig() != null) {
+            log.info("------------------------- Reshard Config found -----------------------------");
+            final Queue<ReshardOptions> reshardQueue = new LinkedList<>(Arrays.asList(consumerConfig.getReshardConfig().getReshardingFactorCycle()));
```

Comment:
FMI: why convert to a `LinkedList|Queue`? What is the expected gain?

Why is the resharding factor cycle defined as a primitive array instead of a `List`? That decision appears to introduce unnecessary complexity.